### ISSUE TITLE
clang-tidy "misc-redundant-expression" false positive for static_assert() using alignof/sizeof

### ISSUE THREAD

**Issue by emezeske** [llvm/llvm-project#63096]
This code triggers the misc-redundant-expression warning. The `static_assert()` is something I've used to catch bugs where struct `X` has fields added to it without the explicit alignment being updated (and this struct is shared memory with a GPU so I need to know when to go update struct definitions on the GPU side). So the `static_assert` does make sense, and works -- the Clang-Tidy warning even appears when the `static_assert` fails (so the two expressions are definitely not redundant)!

```cpp
struct X alignas(32) {
  int x;
};

static_assert(alignof(X) == sizeof(X));
```

**Comment by PiotrZSL** [comment]
This check uses some "hash calculation", most probably issue is there.

**Comment by emezeske** [comment]
w00t thanks!
